.gitlab-ci.yml

```diff
@@ -212,6 +212,24 @@ run_security_scan_test:
     - SNYK_TOKEN=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.snyk_token --with-decryption --query "Parameter.Value" --out text)
       snyk monitor --project-name=datadog-agent-gopkg.lock --file=Gopkg.lock
 
+# check consistency of Gopkg.lock
+check_dep_gopkglock_consistency:
+  stage: source_test
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64:latest
+  tags: [ "runner:main", "size:large" ]
+  allow_failure: true # FIXME as soon as Gopkg.lock is consistent
+  before_script:
+    - cd $SRC_PATH
+    - pip install --upgrade --ignore-installed pip setuptools
+    - pip install -r requirements.txt
+  script:
+    - inv -e deps --verbose --dep-no-vendor
+    - git_diff="$(git diff-files -p Gopkg.lock)"
+    - git_diff_number="$(git diff-files Gopkg.lock | wc -l)"
+    # if there's a change on Gopkg.lock, that means it's not in sync and should be updated: print error message and fail.
+    - test "0" -eq "$git_diff_number" || printf "Gopkg.lock is not in sync with Gopkg.toml and project imports. Please run 'inv deps' and commit the change on Gopkg.lock. Diff:\n$git_diff\n"
+    - test "0" -eq "$git_diff_number"
```

Comment:
Maybe:
```
    - test "0" -eq "$git_diff_number" || (printf "Gopkg.lock is not in sync with Gopkg.toml and project imports. Please run 'inv deps' and commit the change on Gopkg.lock. Diff:\n$git_diff\n" && return 1)
```
That way we can remove the extra `test` command.

Reply:
hmm, I think keeping this line short is better, I don't mind the repetition here :)